cli/schemas/config-file.json

```diff
@@ -0,0 +1,366 @@
+        },
+        "allowUnusedLabels": {
+          "description": "Disable error reporting for unused labels.",
+          "type": "boolean",
+          "default": false,
+          "markdownDescription": "Disable error reporting for unused labels.\n\nSee more: https://www.typescriptlang.org/tsconfig#allowUnusedLabels"
+        },
+        "checkJs": {
+          "description": "Enable error reporting in type-checked JavaScript files.",
+          "type": "boolean",
+          "default": false,
+          "markdownDescription": "Enable error reporting in type-checked JavaScript files.\n\nSee more: https://www.typescriptlang.org/tsconfig#checkJs"
+        },
+        "experimentalDecorators": {
+          "description": "Enable experimental support for TC39 stage 2 draft decorators.",
+          "type": "boolean",
+          "default": true,
+          "markdownDescription": "Enable experimental support for TC39 stage 2 draft decorators.\n\nSee more: https://www.typescriptlang.org/tsconfig#experimentalDecorators"
+        },
+        "jsx": {
+          "description": "Specify what JSX code is generated.",
+          "default": "react",
+          "enum": [
+            "preserve",
+            "react",
+            "react-jsx",
+            "react-jsxdev",
+            "react-native"
+          ]
+        },
+        "jsxFactory": {
+          "description": "Specify the JSX factory function used when targeting React JSX emit, e.g. 'React.createElement' or 'h'",
+          "type": "string",
+          "default": "React.createElement",
+          "markdownDescription": "Specify the JSX factory function used when targeting React JSX emit, e.g. 'React.createElement' or 'h'\n\nSee more: https://www.typescriptlang.org/tsconfig#jsxFactory"
+        },
+        "jsxFragmentFactory": {
+          "description": "Specify the JSX Fragment reference used for fragments when targeting React JSX emit e.g. 'React.Fragment' or 'Fragment'.",
+          "type": "string",
+          "default": "React.Fragment",
+          "markdownDescription": "Specify the JSX Fragment reference used for fragments when targeting React JSX emit e.g. 'React.Fragment' or 'Fragment'.\n\nSee more: https://www.typescriptlang.org/tsconfig#jsxFragmentFactory"
+        },
+        "keyofStringsOnly": {
+          "description": "Make keyof only return strings instead of string, numbers or symbols. Legacy option.",
+          "type": "boolean",
+          "default": false,
+          "markdownDescription": "Make keyof only return strings instead of string, numbers or symbols. Legacy option.\n\nSee more: https://www.typescriptlang.org/tsconfig#keyofStringsOnly"
+        },
+        "lib": {
+          "description": "Specify a set of bundled library declaration files that describe the target runtime environment.",
+          "type": "array",
+          "uniqueItems": true,
+          "default": "deno.window",
+          "items": {
+            "type": "string",
+            "anyOf": [
```

Comment:
I would just drop this and keep it as a string... it is actually really annoying the behaviour in vscode, because tsc issues diagnostics for this and we now mute them, but the schema still has them, so people get "warnings" in vscode that we can't get rid of because it doesn't "match" the schema.  We will end up issuing diagnostics for "bad" or "incorrect" values in config options, so feel free to eliminate any complex "soft" values like this that would be better served by LSP diagnostics, and focus on "structural" validation of the config file.